client/state/sharing/keyring/actions.js

```diff
@@ -48,3 +49,30 @@ export const deleteKeyringConnection = ( connection ) => ( dispatch ) => dispatc
 	type: KEYRING_CONNECTION_DELETE,
 	connection,
 } );
+
+/**
+ * Triggers a network request to delete a Keyring connection from the server-side.
+ *
+ * @param  {Object} connection         Connection to be deleted.
+ * @param  {Number} connection.ID      ID of the connection to be deleted.
+ * @param  {String} connection.label   Name of the service that was connected.
+ * @return {Function}                  Action thunk
+ */
+export function deleteStoredKeyringConnection( connection ) {
```

Comment:
Not sure I understand the purpose of this; what's unique about handling Instagram connections that we need this and a new action type?